app/models/comment.rb

```diff
@@ -7,6 +7,20 @@ class Comment < ActiveRecord::Base
   scope :visible, -> { where(confirmed: true, hidden: false) }
   scope :in_past_week, -> { where("created_at > ?", 7.days.ago) }
 
+  scope :visible_with_unique_emails_for_date, ->(date) {
+    visible.where("date(created_at) = ?", date).group(:email)
+  }
+
+  scope :by_first_time_commenters_for_date, ->(date) {
+    visible_with_unique_emails_for_date(date)
+    .select {|c| where("email = ? AND created_at < ?", c.email, c.created_at.to_date).empty? }
```

Comment:
These will result in dozens of queries for each page view which is obviously not good. Unfortunately I can't think of a better way to do this right now :-/